tests/cancun/eip6780_selfdestruct/test_create2_collision_selfdestruct.py

```diff
@@ -0,0 +1,129 @@
+"""
+Suicide scenario requested test
+https://github.com/ethereum/execution-spec-tests/issues/381
+"""
+
+import pytest
+
+from ethereum_test_forks import Fork
+from ethereum_test_tools import (
+    Account,
+    Environment,
+    Initcode,
+    StateTestFiller,
+    TestAddress,
+    Transaction,
+    compute_create2_address,
+    to_address,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-6780.md"
+REFERENCE_SPEC_VERSION = "2f8299df31bb8173618901a03a8366a3183479b0"
+
+
+@pytest.fixture
+def env():  # noqa: D103
+    return Environment(
+        coinbase="0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba",
+        difficulty=0x020000,
+        gas_limit=71794957647893862,
+        number=1,
+        timestamp=1000,
+    )
+
+
+@pytest.mark.valid_from("Merge")
+@pytest.mark.parametrize(
+    "create2_dest_already_in_state",
+    (False, True),
+)
+def test_create2_collision_selfdestruct(
+    env: Environment,
+    fork: Fork,
+    create2_dest_already_in_state: bool,
+    state_test: StateTestFiller,
+):
+    """Dynamic Create2->Suicide->Create2 collision scenario:"""
+    """
+    Perform a CREATE2, make sure that the initcode sets at least a couple of storage keys,
+    then on a different call, in the same tx, perform a self-destruct.
+    Then:
+        a) on the same tx, attempt to recreate the contract   <=== Covered int this test
+        b) on a different tx, attempt to recreate the contract
+    Perform a CREATE2, make sure that the initcode sets at least a couple of storage keys,
+    then in a different tx, perform a self-destruct.
+    Then:
+        a) on the same tx, attempt to recreate the contract
+        b) on a different tx, attempt to recreate the contract
+    Verify that the test case described
+    in https://wiki.hyperledger.org/pages/viewpage.action?pageId=117440824 is covered
+    """
+
+    address_to = to_address(0x0600)
+    address_s = to_address(0x0512)
+    suicide_d = to_address(0x03E8)
+
+    initcode = Initcode(
+        deploy_code=Op.SELFDESTRUCT(Op.PUSH20(suicide_d)),
+        initcode_prefix=Op.SSTORE(1, 1) + Op.CALL(Op.GAS(), Op.PUSH20(address_s), 0, 0, 0, 0, 0),
+    )
+    create2_address = compute_create2_address(address_to, 0, initcode)
```

Comment:
```suggestion
    create2_salt = 1
    create2_address = compute_create2_address(address_to, create2_salt, initcode)
```

Added a variable here because it seems that we were using different salts on the address calculation and on the actual creation, so a variable here is just a nice way to make sure this is not an issue.